include/BSplineX/bspline/bspline.hpp

```diff
@@ -124,16 +117,17 @@ class BSpline
    */
   BSpline &operator=(BSpline const &other)
   {
-    DEBUG_LOG_CALL();
     if (this == &other)
     {
       return *this;
     }
 
-    knots          = other.knots;
-    control_points = other.control_points;
-    degree         = other.degree;
-    support        = other.support;
+    this->knots          = other.knots;
+    this->control_points = other.control_points;
+    this->degree         = other.degree;
+    this->support        = other.support;
+    this->derivative_ptr = nullptr;
```

Comment:
We can make a deep copy of the unique ptr to copy the derivative.

```suggestion
    this->derivative_ptr =
        other->derivative ? std::unique_ptr<BSpline>{new BSpline{*other->derivative}} : nullptr;

```